packages/crayons-core/src/components/form/form.tsx

```diff
@@ -731,15 +805,30 @@ export class Form {
       ) || {};
   }
 
+  private _getVisibleFields = (): any[] => {
+    let fields = [];
+    if (
+      this.formSchemaState &&
+      Object.keys(this.formSchemaState).length > 0 &&
+      Array.isArray(this.formSchemaState.fields)
+    ) {
+      fields = this.formSchemaState.fields.filter(
+        (field: any) => !this.removeElementFromDomOnHide || !field.hidden
+      );
+    }
+
+    return fields;
+  };
+
   render() {
     const utils: FormUtils = this.composedUtils();
+    const fields: any[] = this._getVisibleFields();
 
     return (
       <form id={`form-${this.formId}`} {...utils.formProps}>
-        {this.formSchemaState &&
-        Object.keys(this.formSchemaState).length > 0 ? (
-          this.formSchemaState?.fields
-            ?.sort((a, b) => a.position - b.position)
+        {fields.length ? (
+          fields
+            .sort((a, b) => a.position - b.position)
             .map((field) => {
               return (
                 this.shouldRenderFormControl(field) && (
```

Comment:
You can use shouldRenderFormControl method to filter the fields instead of `_getVisibleFields` . U can add this logic inside that method